client/me/account/index.jsx

```diff
@@ -210,6 +210,13 @@ module.exports = React.createClass( {
 		} );
 	},
 
+	onSiteSelect( siteSlug ) {
+		let selectedSite = sites.getSite( siteSlug );
+		if ( sites.getSite( siteSlug ) ) {
+			this.props.userSettings.updateSetting( 'primary_site_ID', selectedSite.ID );
```

Comment:
Do we not have a Flux-y action to update the primary site? This looks dangerously entangled here.